### PR TITLE
update cookbook for issue #1269

### LINKED ISSUE #1269: Can't add progress indicator per example in Typescript
I'm trying to use the progress indicator example in my Typescript code, and I get an undefined variable error on global `shepherd` in the `show()` method (tsc shows the error but if you try to run anyway, the var `shepherd` is undefined at runtime too). I'm not sure how to get to the global current tour object with Typescript.

Here's a codesandbox to repro: https://codesandbox.io/s/peaceful-andras-u0ein?file=/src/index.ts

### PR DESCRIPTION
**Pull request by EmNicholson93**
Update for the cookbook directions for the following [issue](https://github.com/shipshapecode/shepherd/issues/1269)

Closes #1269

### CODE REVIEW

File: docs-src/tutorials/04-cookbook.md [hunk 1]
```diff
@@ -54,8 +54,15 @@ being show.
...
-    const header = currentStepElement.querySelector('.shepherd-header');
+    const currentStep = Shepherd.activeTour?.getCurrentStep()
+    if (!currentStep)
+      return
+    const currentStepElement = currentStep.getElement()
+    if (!currentStepElement)
+      return
+    const header = currentStepElement.querySelector('.shepherd-header')
+    if (!header)
+      return
```
Comment (rwwagner90) [inline docs-src/tutorials/04-cookbook.md:65]:
```suggestion
    const currentStep = Shepherd.activeTour?.getCurrentStep();
    const currentStepElement = currentStep?.getElement();
    const header = currentStepElement?.querySelector('.shepherd-header');
```

I think optional chaining looks a lot cleaner here than returning if each thing doesn't exist. What do you think?

File: docs-src/tutorials/04-cookbook.md [hunk 2]
```diff
@@ -54,8 +54,15 @@ being show.
...
     const progress = document.createElement('span');
     progress.style['margin-right'] = '315px';
     progress.innerText = `${shepherd.steps.indexOf(shepherd.currentStep) + 1}/${shepherd.steps.length}`;
```
Comment (rwwagner90) [inline docs-src/tutorials/04-cookbook.md:68]:
```suggestion
    progress.innerText = `${Shepherd.activeTour?.steps.indexOf(currentStep) + 1}/${Shepherd.activeTour?.steps.length}`;
```

Comment (rwwagner90) [reply docs-src/tutorials/04-cookbook.md:68]: Not sure if this works or not, but guessing we need to update it?

Comment (rwwagner90) [reply docs-src/tutorials/04-cookbook.md:68]: We'll need to do `header?.insertBefore` as well to use optional chaining there, etc. Probably should do a quick pass to make sure we've done that everywhere we can.

**Review by rwwagner90 — CHANGES_REQUESTED** [review]
Thanks for looking into this! Left some suggestions around how we could use optional chaining to make this cleaner, at least in my opinion. Let me know what you think!

File: docs-src/tutorials/04-cookbook.md (hunk 2 above)
Comment (EmNicholson93) [reply docs-src/tutorials/04-cookbook.md:62]: - added optional chaining to the header/footer in the examples

Is the progress indicator used within the example? I'm not finding step options implicating the show option but I could totally be missing that.

Comment (rwwagner90) [reply docs-src/tutorials/04-cookbook.md]: @EmNicholson93 no, it is not used in the example. Perhaps we should build a small set of examples that take what we have in the cookbook and actually run it? We could then allow users to see various things they could possibly do.

File: docs-src/tutorials/04-cookbook.md [hunk 3]
```diff
@@ -54,12 +54,13 @@ being show.
...
-    const currentStepElement = shepherd.currentStep.el;
-    const header = currentStepElement.querySelector('.shepherd-header');
+    const currentStep = Shepherd.activeTour?.getCurrentStep();
+    const currentStepElement = currentStep?.getElement();
+    const header = currentStepElement?.querySelector('.shepherd-header');
     const progress = document.createElement('span');
     progress.style['margin-right'] = '315px';
-    progress.innerText = `${shepherd.steps.indexOf(shepherd.currentStep) + 1}/${shepherd.steps.length}`;
-    header.insertBefore(progress, currentStepElement.querySelector('.shepherd-cancel-icon'));        
+    progress.innerText = `${shepherd.activeTour?.steps.indexOf(shepherd.currentStep) + 1}/${shepherd.steps.length}`;
```
Comment (rwwagner90) [inline docs-src/tutorials/04-cookbook.md:62]:
```suggestion
    progress.innerText = `${Shepherd.activeTour?.steps.indexOf(currentStep) + 1}/${Shepherd.activeTour?.steps.length}`;
```

File: docs-src/tutorials/04-cookbook.md [hunk 4]
```diff
@@ -69,12 +70,13 @@ Another example, for anyone who wants to add progress indicators to the footer.
...
-    const currentStepElement = shepherd.currentStep.el;
-    const footer = currentStepElement.querySelector('.shepherd-footer');
+    const currentStep = Shepherd.activeTour?.getCurrentStep();
+    const currentStepElement = currentStep?.getElement();
+    const footer = currentStepElement?.querySelector('.shepherd-footer');
     const progress = document.createElement('span');
     progress.className = 'shepherd-progress';
-    progress.innerText = `${shepherd.steps.indexOf(shepherd.currentStep) + 1} of ${shepherd.steps.length}`;
-    footer.insertBefore(progress, currentStepElement.querySelector('.shepherd-button:last-child'));
+    progress.innerText = `${shepherd.activeTour?.steps.indexOf(shepherd.currentStep) + 1} of ${shepherd.steps.length}`;
```
Comment (rwwagner90) [inline docs-src/tutorials/04-cookbook.md:78]:
```suggestion
    progress.innerText = `${Shepherd.activeTour?.steps.indexOf(currentStep) + 1} of ${Shepherd.activeTour?.steps.length}`;
```

**Review by rwwagner90 — CHANGES_REQUESTED** [review]

**Review by rwwagner90 — APPROVED** [review]

**Pull request merged**